packages/ts-endpoint/src/Endpoint/Endpoint.ts

```diff
@@ -44,14 +26,15 @@ export interface Endpoint<
   H extends { [k: string]: t.Type<any, any, any> } | undefined = undefined,
   Q extends { [k: string]: t.Type<any, any, any> } | undefined = undefined,
   B extends t.Type<any, any, any> | undefined = undefined,
-  P extends { [k: string]: t.Type<any, any, any> } | undefined = undefined
+  P extends { [k: string]: t.Type<any, any, any> } | undefined = undefined,
+  E extends Array<EndpointError<any, any>> | undefined = undefined
```

Comment:
Are you sure you want an array of tuples to define the errors?
I see the benefits when you need to specify different error shape based on error code, but I think most of the times you just need to define the error for all status codes.

Reply:
I was unsure about thin until I had the need the other day... That's why I made the property optional, so that is you don't need it you can just skip it.
I think that it is just added detail that can be used by the `ts-endpoint-express` and `ts-endpoint-browser` packages in case they need it

Reply:
I think it would be better to leave the errors shape out of endpoint definition and add it only to `ts-endpoint-express`.

Something like would work (I think):
```js
AddEndpoint(router, {
 errorHandler: (err) => { },
})((req) => { ... })
```

If you want to keep it in the endpoint definition I would use a function that map error to desired shape, something like: 

```js
const errorHandler = (err: Error): MyError => {...}
```

I don't see why we would need so many different shapes for the errors based on status code

Reply:
> AddEndpoint(router, {
 errorHandler: (err) => { },
})((req) => { ... })

this is actually almost exactly what I put on `ts-endpoint-express`, so I agree :100: .

The reason for having errors defined directly on `Endpoint` is because sometimes the contract between client and server includes some "expected" failures (like a 404 in case of a "search" API call or a 400 in case of backend form validation).

Some of these expected errors have a specific shape that does not fit into a generic one (like Error). For instance, the backend validation may have a body with a `errors` field which is an array of `[fildName: string, validationError: string]`.

This is why lately I felt the need to include in the Endpoint contract the shape of these "expected" failures so that I can have some static guarantees when I pattern-match on the error.

What do you think?

Reply:
I think we could give it a chance :)